pkg/kademlia/routing.go

```diff
@@ -321,3 +323,73 @@ func (rt *RoutingTable) ConnSuccess(ctx context.Context, node *pb.Node) {
 		zap.L().Debug("connection success error:", zap.Error(err))
 	}
 }
+
+func splitBucket(bIDs []bucketID, bitDepth int) ([]bucketID,[]bucketID) {
+	var b0 []bucketID
+	var b1 []bucketID
+
+	for _, bID := range bIDs {
+		byteDepth := bitDepth / 8
+		bitOffset := bitDepth % 8
+		power := uint(7 - bitOffset)
+		bitMask := byte(1 << power)
+		b := bID[byteDepth]
+		if b&bitMask > 0 {
+			b1 = append(b1,bID)
+		} else {
+			b0 = append(b0,bID)
+		}
+	}
+	return b0,b1
+}
+
+func (rt *RoutingTable) BufferedGraph(buf *bytes.Buffer) {
+	buf.Write([]byte("digraph{\nnode [shape=box];edge [dir=none];\n"))
+
+	ids, _ := rt.GetBucketIds()
+	var bucketids []bucketID
+	for _,n := range ids {
+		bucketids = append(bucketids, keyToBucketID(n))
+	}
+
+	rt.addBucketsToGraph(bucketids, 0, buf, "")
+
+	buf.Write([]byte("}\n"))
+
+}
+
+func (rt *RoutingTable) addLeafBucketToGraph(b bucketID, buf *bytes.Buffer, prefix string) {
+	fmt.Fprintf(buf, "b%s [label=<<b><font point-size=\"18\">%s</font></b><br />\n<i>routing:</i><br align=\"left\"/>", prefix, prefix)
+		
+	nodes, _ := rt.getNodeIDsWithinKBucket(b)
+	for _, n := range nodes {
+		fmt.Fprintf(buf, "  %s<br align=\"left\" />", hex.EncodeToString(n[:]))
```

Comment:
Instead of `hex.EncodeToString(n[:])` try calling `n.String()`, which will display the node ID in the expected way.

Reply:
Let's also display the node address together with the node id. I think this will make easier to analyze the routing table because the node ids themselves are hard to comprehend.

The node address can be added in paranthesis after the node id, i.e. `<node_id> (<node_address>)`.